ui/selectors/selectors.js

```diff
@@ -1757,6 +1762,25 @@ export function getCustomTokenAmount(state) {
   return state.appState.customTokenAmount;
 }
 
+export function getUpdatedAndSortedAccounts(state) {
+  const accounts = getMetaMaskAccountsOrdered(state);
+  const pinnedAccounts = getPinnedAccountsList(state);
+  accounts.forEach((account) => {
+    account.pinned = Boolean(pinnedAccounts?.includes(account.address));
```

Comment:
would it not be possible to include a pinned property on the "main" accounts list so we dont need to maintain two lists? and reconcile them every time? @k-g-j is this something that we can do with the accounts controller?

Reply:
Yes the plan is to do that in the future, but not block merging this PR because of it.